File: src/main/java/com/example/seollyongbackend/repository/UserRepository.java
package com.example.seollyongbackend.repository;

import com.example.seollyongbackend.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    User findByUsername(String username);
    User findByNickname(String nickname);
}

File: src/main/kotlin/com/example/seollyongdanbackend/repository/UserRepository.kt
package com.example.seollyongdanbackend.repository

import com.example.seollyongdanbackend.entity.User
import org.springframework.data.jpa.repository.JpaRepository //기본적인 CRUD 기능 제공
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {

    // 아이디 중복 확인용
    fun findByUsername(username: String): User?

    // 닉네임 중복 확인용
    fun findByNickname(nickname: String): User?

}